### PR TITLE
fix:  active campaign: message.context not sent and tags sent in non array format

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> This PR includes the changes needed to be done for active campaign destination to solve bugsnag issue asociated to `msgTags`
1. Introducing Optional Chaining at multiple places for `message.context.traits` so it won't not go on to finding `traits` in `message.context.traits` when `context` is not present hence won’t throw error.
2. Correcting an if condition to check if the `msgTags` are present and also if they are in array format or not

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by mihir-4116 — DISMISSED** [review]
LGTM

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/active_campaign/transform.js [hunk 1]
```diff
@@ -124,9 +124,8 @@ const customTagProcessor = async (message, category, destination, contactId) =>
     if (parseInt(get(res, TOTAL_RECORDS_KEY), 10) > 100) {
       const limit = Math.floor(parseInt(get(res, TOTAL_RECORDS_KEY), 10) / 100);
       for (let i = 0; i < limit; i += 1) {
-        endpoint = `${destination.Config.apiUrl}${category.tagEndPoint}?limit=100&offset=${
-          100 * (i + 1)
-        }`;
+        endpoint = `${destination.Config.apiUrl}${category.tagEndPoint}?limit=100&offset=${100 * (i + 1)
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:127]:
```suggestion
endpoint = `${destination.Config.apiUrl}${category.tagEndPoint}?limit=100&offset=${100 * (i + 1)}`
```

Comment (sanpj2292) [reply src/v0/destinations/active_campaign/transform.js:127]: Keep this in a single line

File: src/v0/destinations/active_campaign/transform.js [hunk 2]
```diff
@@ -218,7 +215,7 @@ const customFieldProcessor = async (message, category, destination) => {
   // Step - 2
   // Get the existing field data from dest and store it in responseStaging
   // Ref - https://developers.activecampaign.com/reference/retrieve-fields
-  let endpoint = `${destination.Config.apiUrl}${`${category.fieldEndPoint}?limit=100`}`;
+  let endpoint = `${destination.Config.apiUrl}${`${category.fieldEndPoint}?limit=100`} `;
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:218]:
```suggestion
  let endpoint = `${destination.Config.apiUrl}${category.fieldEndPoint}?limit=100 `;
```

File: src/v0/destinations/active_campaign/transform.js [hunk 3]
```diff
@@ -234,9 +231,8 @@ const customFieldProcessor = async (message, category, destination) => {
   const limit = Math.floor(parseInt(get(res, TOTAL_RECORDS_KEY), 10) / 100);
   if (parseInt(get(res, TOTAL_RECORDS_KEY), 10) > 100) {
     for (let i = 0; i < limit; i += 1) {
-      endpoint = `${destination.Config.apiUrl}${category.fieldEndPoint}?limit=100&offset=${
-        100 * (i + 1)
-      }`;
+      endpoint = `${destination.Config.apiUrl}${category.fieldEndPoint}?limit = 100 & offset=${100 * (i + 1)
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:234]:
```suggestion
      endpoint = `${destination.Config.apiUrl}${category.fieldEndPoint}?limit=100&offset=${100 * (i + 1)}`
```

File: src/v0/destinations/active_campaign/transform.js [hunk 4]
```diff
@@ -286,7 +282,7 @@ const customFieldProcessor = async (message, category, destination) => {
     if (Array.isArray(fieldInfo[key])) {
       fPayload = '||';
       fieldInfo[key].map((fv) => {
-        fPayload = `${fPayload}${fv}||`;
+        fPayload = `${fPayload}${fv}|| `;
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:285]:
```suggestion
        fPayload = `${fPayload}${fv}||`;
```
Please check if this is valid

File: src/v0/destinations/active_campaign/transform.js [hunk 5]
```diff
@@ -323,7 +319,7 @@ const customListProcessor = async (message, category, destination, contactId) =>
   const promises = [];
   listArr.map(async (li) => {
     if (li.status === 'subscribe' || li.status === 'unsubscribe') {
-      const endpoint = `${destination.Config.apiUrl}${category.mergeListWithContactUrl}`;
+      const endpoint = `${destination.Config.apiUrl}${category.mergeListWithContactUrl} `;
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:322]:
```suggestion
      const endpoint = `${destination.Config.apiUrl}${category.mergeListWithContactUrl}`;
```

File: src/v0/destinations/active_campaign/transform.js [hunk 6]
```diff
@@ -384,7 +380,7 @@ const screenRequestHandler = async (message, category, destination) => {
   // Retrieve All events from destination
   // https://developers.activecampaign.com/reference/list-all-event-types
   let res;
-  let endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint}`;
+  let endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint} `;
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:383]:
```suggestion
  let endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint}`;
```

File: src/v0/destinations/active_campaign/transform.js [hunk 7]
```diff
@@ -410,7 +406,7 @@ const screenRequestHandler = async (message, category, destination) => {
   // Ref - https://developers.activecampaign.com/reference/create-a-new-event-name-only
   if (!events.includes(message.event)) {
     // Create the event
-    endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint}`;
+    endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint} `;
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:409]:
```suggestion
    endpoint = `${destination.Config.apiUrl}${category.getEventEndPoint}`;
```

File: src/v0/destinations/active_campaign/transform.js (hunk 4 above)
Comment (anantjain45823) [reply src/v0/destinations/active_campaign/transform.js:285]: Yes this is valid case. Checked

**Review by mihir-4116 — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**